src/NuGet.Clients/NuGet.PackageManagement.UI/SolutionPickerWindow.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Windows;
+using NuGet.PackageManagement.VisualStudio;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class SolutionPickerWindow : VsDialogWindow
+    {
+        public SolutionPickerWindow(SolutionPickerViewModel viewModel)
+        {
+            if (viewModel == null)
+            {
+                throw new ArgumentNullException(nameof(viewModel));
+            }
+
+            Content = new SolutionPickerView(viewModel);
+
+            EventHandler handler = null;
+            handler = (s, e) =>
+            {
+                viewModel.SolutionClicked -= handler;
+                DialogResult = true;
+            };
+            viewModel.SolutionClicked += handler;
```

Comment:
Create a an event handler method that isn't anonymous. Don't unhook the handler in the method itself. Instead unregister it in the Dispose method of this class.

Also shouldn't you be calling close here?